### PR TITLE
GH-541 Adds Toolbar Widget

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith @jbrundage @dtsnell4 

Refactored code ... suggestions welcomed

Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
http://rawgit.com/mlzummo/Visualization/GH_541_TOOLBAR_WIDGET/demos/dermatology.html?src/layout/Toolbar

**Comment by mlzummo** [comment]
works locally, but on rawgit the widget doesnt render fast enough and the widget height is still 0 when twTop is calculated causing an alignment issue

File: src/layout/Toolbar.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        this.widgetArr([
+            new FormInput().name("textbox-test").label("Only Alpha").type("textbox").value("SomeString"),
+            new FormInput().name("checkbox-test").label("Checkbox Test").type("checkbox").value(true),
+            new Slider().name("slider-test").label("Slider Test").value(66),
+            new FormInput().type("button").value("button 2").label("button 2").name("button2"),
+            new Icon().testData()
+        ]);
+        this.toolbarAnnotations([
+            {
+                type: "button",
```
Comment (mlzummo) [inline src/layout/Toolbar.js]: the type property can be removed now

**Comment by GordonSmith** [comment]
Commits need to be rolled up.

File: src/layout/Toolbar.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        ]);
+        return this;
+    };
+
+    Toolbar.prototype.enter = function (domNode, element) {
+        HTMLWidget.prototype.enter.apply(this, arguments);
+        this._toolbarContainer = element
+                .append("div")
+                .attr("class", "toolbar-container")
+                //.style("position","relative")
```
Comment (GordonSmith) [inline src/layout/Toolbar.js]: remove commented code.

**Comment by GordonSmith** [comment]
Closing until it is ready to be pulled.

**Pull request closed without merging**